src/main/java/com/google/devtools/build/lib/packages/BazelStarlarkEnvironment.java

```diff
@@ -338,17 +341,57 @@ private static boolean injectionApplies(String key, Map<String, Boolean> overrid
    * documentation for {@code --experimental_builtins_injection_override}. Non-injected symbols must
    * still obey the above constraints.
    *
-   * @see StarlarkBuiltinsFunction
+   * @see com.google.devtools.build.lib.skyframe.StarlarkBuiltinsFunction
    */
   public ImmutableMap<String, Object> createBuildBzlEnvUsingInjection(
       Map<String, Object> exportedToplevels,
       Map<String, Object> exportedRules,
       List<String> overridesList)
       throws InjectionException {
+    return createBuildBzlEnvUsingInjection(
+        exportedToplevels, exportedRules, overridesList, uninjectedBuildBzlNativeBindings);
+  }
+
+  /**
+   * Constructs an environment for a WORKSPACE-loaded bzl file based on the default environment, the
+   * maps corresponding to the {@code exported_toplevels} and {@code exported_rules} dicts, and the
+   * value of {@code --experimental_builtins_injection_override}.
+   *
+   * @see com.google.devtools.build.lib.skyframe.StarlarkBuiltinsFunction
+   */
+  public ImmutableMap<String, Object> createWorkspaceBzlEnvUsingInjection(
+      Map<String, Object> exportedToplevels,
+      Map<String, Object> exportedRules,
+      List<String> overridesList)
+      throws InjectionException {
+    return createBuildBzlEnvUsingInjection(
+        exportedToplevels, exportedRules, overridesList, workspaceBzlNativeBindings);
+  }
+
+  private ImmutableMap<String, Object> createBuildBzlEnvUsingInjection(
+      Map<String, Object> exportedToplevels,
+      Map<String, Object> exportedRules,
+      List<String> overridesList,
+      Map<String, Object> nativeBase)
+      throws InjectionException {
     Map<String, Boolean> overridesMap = parseInjectionOverridesList(overridesList);
 
+    Map<String, Object> env = new HashMap<>(ruleClassProvider.getEnvironment());
```

Comment:
Using `ruleClassProvider.getEnvironment()` looks like a semantic change. Before it was `uninjectedBuildBzlEnv`.

Reply:
Yes—I think what we should do is remove support for disabling builtins injections, since that can't possibly result in a working Bazel anymore.